src/pages/index.js

```diff
@@ -60,28 +107,122 @@ document.addEventListener("DOMContentLoaded", function () {
     addFormValidator.resetValidation();
   });
 
-  newCardList.renderItems();
+  constants.avatar.addEventListener("click", () => {
+    avatarPopup.open();
+    addFormValidator.resetValidation();
+  });
+
+  constants.avatar.addEventListener("mousemove", () => {
+    constants.avatarButton.classList.replace(
+      "profile__avatar-button_disabled",
+      "profile__avatar-button"
+    );
+  });
+});
+
+constants.avatar.addEventListener("mouseout", () => {
+  constants.avatarButton.classList.replace(
+    "profile__avatar-button",
+    "profile__avatar-button_disabled"
+  );
 });
 
 /* -------------------------------------------------------------------------- */
 /*                                  Functions                                 */
 /* -------------------------------------------------------------------------- */
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    (id) => handleDeleteClick(id),
+    (id) => handleLikeClick(id)
+  );
   return card.getView();
 }
 
-function handleImageClick(name, link) {
-  newPopupImage.open(name, link);
+function handleImageClick(cardData) {
+  newPopupImage.open(cardData);
+}
+
+function handleDeleteClick(card) {
+  popupAffirm.open();
+  popupAffirm.setSubmitAction(() => {
+    api
+      .deleteCards(card._id)
+      .then(() => {
+        popupAffirm.close();
+        card.handleDeleteCard();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+function handleLikeClick(card) {
+  if (card.isLiked === true) {
+    api
+      .deleteCardLike(card._id)
+      .then((res) => {
+        card.isLiked = false;
+        card.handleLikeIcon();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .putCardLike(card._id)
+      .then((res) => {
+        card.isLiked = true;
+        card.handleLikeIcon();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
 }
 
 function handleProfileFormSubmit(formData) {
-  useInfo.setUserInfo(formData.title, formData.description);
-  editPopup.close();
+  constants.editProfileSubmitButton.textContent = "Saving...";
+  api
+    .updateProfileInfo(formData.title, formData.description)
+    .then((res) => {
+      useInfo.setUserInfo(res.name, res.about);
+      editPopup.close();
+      constants.editProfileSubmitButton.textContent = "Save";
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 
 function handleCardFormSubmit(name, link) {
-  newCardList.addItem(name, link);
-  cardPopup.close();
-  constants.addCardForm.reset();
+  constants.addCardSubmitButton.textContent = "Saving...";
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```